piquasso/_math/combinatorics.py

```diff
@@ -106,3 +106,204 @@ def partitions(boxes, particles, out=None):
             separators[j] = separators[j - 1] + 1
 
     return result
+
+
+def partitions_bounded_k(boxes, particles, constrained_boxes, max_per_box, k_limit):
+    """
+    Return all weak compositions x of 'particles' into 'boxes' such that:
+      - for each constrained box m = constrained_boxes[i]:
+            x[m] <= max_per_box[i]
```

Comment:
What is `x[m]`? x is all weak compositions if I understand correctly, then m is a constrained box, why would we index into the weak compositions with a constrained box? What does that give us?

Even though everything seems to be explained, I get confused about what each of the names actually means.

Reply:
I think here, or in the file of the SLOS/Clifford & Clifford algo is implemented works. :+1: